### PR TITLE
Remove deprecated `:example_group` example metadata sub-hash

### PR DESCRIPTION
**Pull request by pirj**
See:
 - https://github.com/rspec/rspec-core/pull/1490
 - https://github.com/rspec/rspec-core/commit/9ffcc3873365730e12a2a5b6883db9da06fd1eda

### CODE REVIEW

**Comment by pirj** [comment]
@JonRowe I would really appreciate your help here. Specs are failing and I guess I did something wrong.

**Comment by pirj** [comment]
@JonRowe I appreciate your advice here. I might have been too radical with the changes (again).
What is the correct approach here?

File: lib/rspec/core/metadata.rb [hunk 1]
```diff
@@ -326,7 +275,7 @@ def full_description
         :description,
         :description_args,
         :described_class,
-        :example_group,
+        :example_group, # FIXME: should we also remove this?
```
Comment (pirj) [inline lib/rspec/core/metadata.rb:278]: Highlight. Should we remove this?

Comment (JonRowe) [reply lib/rspec/core/metadata.rb:278]: I don't think so, we still set it and can be used to retrieve data, so I think it needs to stay reserved to preserve any down stream behaviour that uses metadata for other purposes?

File: Changelog.md [hunk 2]
```diff
@@ -13,6 +13,8 @@ Breaking Changes:
 * Raise on usage of metadata on suite-level scopes. (Phil Pirozhkov, #2849)
 * Raise an error when `fail_fast` is configured with
   an unsupported value. (Phil Pirozhkov, #2849)
+* Remove deprecated exposition of example group's metadata
+  in example's metadata `:example_group` subhash. (Phil Pirozhkov, #2851)
```
Comment (JonRowe) [inline Changelog.md:17]: This needs rewording but I havent had enough coffee yet to process how.

**Comment by JonRowe** [comment]
This looks good to me overall

File: Changelog.md (hunk 2 above)
Comment (JonRowe) [inline Changelog.md:17]:
```suggestion
* Remove deprecated access to an example group's metadata through the example.
  (Phil Pirozhkov, #2851)
```

**Review by JonRowe — APPROVED** [review]

File: Changelog.md (hunk 2 above)
Comment (pirj) [reply Changelog.md:17]: Thanks! Much better this way.

**Pull request merged**